### PR TITLE
Remove File interface and rename FileImpl with File

### PR DESCRIPTION
**Pull request by mangalaman93**
Ref: https://rakyll.org/interface-pollution/
Interfaces can be defined by the user if and when necessary.

### CODE REVIEW

File: mmap.go [hunk 1]
```diff
@@ -15,25 +13,8 @@ var (
...
-	Unlock() error
-	Advise(advice int) error
-	Flush(flags int) error
-	Unmap() error
-}
-
-// FileImpl provides abstraction around a memory mapped file
-type FileImpl struct {
+// File provides abstraction around a memory mapped file
+type File struct {
```
Comment (rpip) [inline mmap.go]: Agree with the changes, since the File is primarily used by bigqueue.

Why did you remove the io.ReaderAt and io.WriterAt embeds from the File struct?

Comment (mangalaman93) [reply mmap.go]: You can't have interfaces embedded into a struct. Only an interface can have embedded interface in it. The type `File` still satisfies `io.ReaderAt` and `io.WriterAt` and we still have test for that too here https://github.com/grandecola/mmap/blob/master/mmap_test.go#L550
